ml-agents/mlagents/trainers/tests/test_environments/test_simple.py

```diff
@@ -0,0 +1,256 @@
+import yaml
+import math
+import tempfile
+from typing import Any, Callable, Dict, Optional, List
+
+
+from mlagents.trainers.trainer_controller import TrainerController
+from mlagents.envs.base_unity_environment import BaseUnityEnvironment
+from mlagents.envs.env_manager import EnvManager, StepInfo
+from mlagents.envs.timers import timed
+from mlagents.envs import ActionInfo, BrainInfo, AllBrainInfo, BrainParameters
+from mlagents.envs.communicator_objects import AgentInfoProto
+
+
+BRAIN_NAME = __name__
+OBS_SIZE = 1
+STEP_SIZE = 0.1
+
+
+def clamp(x, min_val, max_val):
+    return max(min_val, min(x, max_val))
+
+
+class Simple1DEnvironment(BaseUnityEnvironment):
+    """
+    Very simple "game" - the agent has a position on [-1, 1], gets a reward of 1 if it reaches 1, and a reward of -1 if
+    it reaches -1. The position is incremented by the action amount (clamped to [-step_size, step_size]).
+    """
+
+    def __init__(self):
+        self._brains: Dict[str, BrainParameters] = {}
+        self._brains[BRAIN_NAME] = BrainParameters(
+            brain_name=BRAIN_NAME,
+            vector_observation_space_size=OBS_SIZE,
+            num_stacked_vector_observations=1,
+            camera_resolutions=[],
+            vector_action_space_size=[1],
+            vector_action_descriptions=["moveDirection"],
+            # vector_action_space_type=0,  # "discrete"
+            vector_action_space_type=1,  # "continuous"
+        )
+
+        # state
+        self.position = 0.0
+        self.step_count = 0
+
+    def step(
+        self,
+        vector_action: Dict[str, Any] = None,
+        memory: Dict[str, Any] = None,
+        text_action: Dict[str, Any] = None,
+        value: Dict[str, Any] = None,
+    ) -> AllBrainInfo:
+        if vector_action:
+            delta = vector_action[BRAIN_NAME][0][0]
+            self.position += clamp(delta, -STEP_SIZE, STEP_SIZE)
+            self.position = clamp(self.position, -1, 1)
+            self.step_count += 1
+        done = self.position >= 1.0 or self.position <= -1.0
+        reward = self.position if done else 0.0
```

Comment:
We should probably add a small time penalty (~-0.001) as well so the agent is incentivized to got to the right position quickly, and change the win threshold accordingly

Reply:
Done.